### PR TITLE
EVG-20568 Add GitHub merge queue subscription type

### PR DESCRIPTION
**Pull request by bsamek**
EVG-20568

### Description

I incorrectly used the GitHub checks subscription type, which is specifically
intended for the [GitHub
Checks](https://docs.devprod.prod.corp.mongodb.com/evergreen/Project-Configuration/Project-and-Distro-Settings#github-checks)
feature. I added a new subscription type for the merge queue.

### Testing

I added the new subscriber type to the existing test framework.

### Documentation

None needed. Internal changes only.

### CODE REVIEW

File: model/notification/notification.go [hunk 1]
```diff
@@ -203,6 +203,17 @@ func (n *Notification) Composer(env evergreen.Environment) (message.Composer, er
 		payload.Ref = sub.Ref
 		return message.NewGithubStatusMessageWithRepo(level.Notice, *payload), nil
 
+	case event.GithubMergeSubscriberType:
+		sub := n.Subscriber.Target.(*event.GithubMergeSubscriber)
+		payload, ok := n.Payload.(*message.GithubStatus)
+		if !ok || payload == nil {
+			return nil, errors.New("github-check payload is invalid")
```
Comment (malikchaya2) [inline model/notification/notification.go:210]: should this be github merge queue?

File: rest/model/subscriber.go [hunk 2]
```diff
@@ -129,6 +144,16 @@ func (s *APISubscriber) ToService() (event.Subscriber, error) {
 		}
 		target = apiModel.ToService()
 
+	case event.GithubMergeSubscriberType:
+		apiModel := APIGithubMergeSubscriber{}
+		if err = mapstructure.Decode(s.Target, &apiModel); err != nil {
+			return event.Subscriber{}, gimlet.ErrorResponse{
+				StatusCode: http.StatusBadRequest,
+				Message:    errors.Wrap(err, "GitHub check subscriber target is malformed").Error(),
```
Comment (malikchaya2) [inline rest/model/subscriber.go:152]: *GitHub merge queue

**Review by malikchaya2 — APPROVED** [review]
Approved with small comments

File: model/notification/notification.go (hunk 1 above)
Comment (bsamek) [reply model/notification/notification.go:210]: fixed

File: rest/model/subscriber.go (hunk 2 above)
Comment (bsamek) [reply rest/model/subscriber.go:152]: fixed

**Review by malikchaya2 — APPROVED** [review]

**Pull request merged**